### PR TITLE
Make cli auth case insensitive

### LINKED ISSUE #100: make CLI authentication case-insensitive
in  #99, we found out that users assume that `authentication: CLI` can also be given as `authentication: cli` which is not the case.

@JCZuurmond... if you're interested 👀 

Spitball idea, but maybe we add an `.upper()` to the this snippet below. something like:
```python
if type_auth.upper() in AZURE_AUTH_FUNCTIONS.keys()
```

https://github.com/dbt-msft/dbt-sqlserver/blob/0d529de2641cc7e3f84bab47e22a9f4c80629bbd/dbt/adapters/sqlserver/connections.py#L281-L288

### PR DESCRIPTION
**Pull request by JCZuurmond**
Fixes #100

### CODE REVIEW

**Comment by mikaelene** [comment]
@swanderz I have not used the CLI. But does this looks good to you?

**Review by swanderz — CHANGES_REQUESTED** [review]
thanks @JCZuurmond! can you add a "under the hood"  bullet to `CHANGELOG.MD` under a `future` release heading?

**Comment by JCZuurmond** [comment]
@swanderz : Is this ok?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,15 @@
 # Changelog
 
+### v0.19.0.3
+
+#### under the hood
+- cli authentication is case insensitive. [#100](https://github.com/dbt-msft/dbt-sqlserver/issues/100)
```
Comment (swanderz) [inline CHANGELOG.md:6]:
```suggestion
- authentication now is case-insensitive and accepts both `CLI` and `cli` as options. [#100](https://github.com/dbt-msft/dbt-sqlserver/issues/100) thanks (@JCZuurmond)[https://github.com/JCZuurmond]
- add unit tests for azure-identity related token fetching
```

**Review by swanderz — APPROVED** [review]

**Comment by JCZuurmond** [comment]
I think the tests are failing for a different reason.

**Comment by JCZuurmond** [comment]
No worries! Take the time you need.

**Pull request merged**